internal/tuf/tuf.go

```diff
@@ -177,6 +181,13 @@ type RootMetadata interface {
 	// network for which the current repository is a controller.
 	// IsController must return true for this to be set.
 	GetNetworkRepositories() []OtherRepository
+
+	// AddHook adds a hook to the metadata file.
+	AddHook(stage, hookName, env string, hashes map[string]gitinterface.Hash, modules, principalIDs []string) error
```

Comment:
Can you add what the parameters do to the docstring? `env` for example isn't very self describing.

Reply:
Are we going with hook? I wonder about using something else in the interface to avoid collision with git hooks.

Reply:
With my proposal for making stage a type, we could pass in `HookStage` type here and improve validation. The `Hooks` structs could also just use stage as the internal map key.